js/resultsPanel.js

```diff
@@ -0,0 +1,83 @@
+class ResultsPanel {
+    constructor(domContainer) {
+      this.container = domContainer;
+      this.generate();
+    }
+  
+    generate() {
+      this.resultsList = makeElement('div', { className: 'results-list' });
+      this.container.append(this.resultsList);
+    }
+  
+    clearAll() {
+      const removeItems = [...this.resultsList.children];
+      removeItems.forEach(li => li.remove());
+    }
+  
+    async addAllRes(data) {
+      let promAllLines = data.map(this.createResultLine);
+      const createdLines = await Promise.all(promAllLines);
+      this.resultsList.append(...createdLines);
+    }
+  
+    addNothingFound() {
+      const itemContainer = makeElement('div', {
+        className: 'result-item-container no-item',
+      });
+  
+      const message = makeElement('div', {
+        className: 'search-result-text',
+        innerText: 'Apologies, no stock found 🤷',
+      });
+  
+      const image = makeElement('img', {
+        className: 'search-result-img',
+        src: '/assets/img/thinking-emoji-trans.gif',
+      });
+  
+      itemContainer.append(image, message);
+      this.resultsList.append(itemContainer);
+    }
+  
+    async createResultLine(resultObj) {
+      const itemContainer = makeElement('div', {
+        className: 'result-item-container',
+      });
+  
+      const profile = await getCompanyProfile(resultObj.symbol);
```

Comment:
altough your logic works which is very good, it's better to have all the login regarding retreving the data from the server in one place (one class, the search class) and the UI logic (everything regarding drawing the HTML and the DOM manipulation) in another class (resultsPanel).